var/spack/repos/builtin/packages/ior/package.py

```diff
@@ -38,17 +37,23 @@ class Ior(Package):
     variant('hdf5',  default=False, description='support IO with HDF5 backend')
     variant('ncmpi', default=False, description='support IO with NCMPI backend')
 
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool', type='build')
+    depends_on('m4', type='build')
     depends_on('mpi')
-    depends_on('hdf5+mpi',   when='+hdf5')
+    depends_on('hdf5+mpi', when='+hdf5')
     depends_on('parallel-netcdf', when='+ncmpi')
 
-    def install(self, spec, prefix):
-        os.system('./bootstrap')
+    @run_before('autoreconf')
+    def bootstrap(self):
+        Executable('./bootstrap')()
 
-        config_args = [
-            'MPICC=%s' % spec['mpi'].prefix.bin + '/mpicc',
-            '--prefix=%s' % prefix,
-        ]
+    def configure_args(self):
+        spec = self.spec
+        config_args = []
+
+        env['CC'] = spec['mpi'].mpicc
```

Comment:
It was MPICC before, but I assume this works as well

Reply:
when I converted to AutotoolsPackage, `CC` is set to `icc` and IOR makefile tries to compile mpi code with `icc` resulting in `mpi.h` found error. Hence above change.